mimic/test/test_ironic.py

```diff
@@ -45,19 +45,114 @@ def setUp(self):
             "power_state", "instance_info", "ports", "name", "driver_internal_info",
             "inspection_finished_at", "inspection_started_at", "clean_step"
         ]
+        self.url = "/ironic/v1/nodes"
+        self.create_request = {
+            "chassis_uuid": str(uuid4()),
+            "driver": "fake",
+            "driver_info": {"cache_image_id": None,
+                            "cache_status": None},
+            "name": "test_node",
+            "properties": {
+                "cpus": "1",
+                "local_gb": "10",
+                "memory_mb": "1024"
+            }
+        }
+
+    def create_node(self, create_request=None):
+        """
+        Create a new node and return node.
+        """
+        request = create_request or self.create_request
+        (response, content) = self.successResultOf(json_request(
+            self, self.root, "POST", self.url, body=json.dumps(request)))
+        self.assertEqual(response.code, 201)
+        return content
 
     def get_nodes(self, postfix=None):
         """
         Get nodes and return content
         """
-        url = "/ironic/v1/nodes"
+        url = self.url
         if postfix:
-            url = "/ironic/v1/nodes" + postfix
+            url = self.url + postfix
         (response, content) = self.successResultOf(json_request(
             self, self.root, "GET", url))
         self.assertEqual(200, response.code)
         return content
 
+    def test_create_node(self):
+        """
+        Create node returns 201 and the newly created node.
+        """
+        new_node = self.create_node()
+        for key in self.create_request.keys():
+            self.assertEqual(self.create_request[key], new_node[key])
+
+    def test_create_node_failure(self):
+        """
+        Create node returns 400 if the request json is not
+        as expected.
+        """
+        response = self.successResultOf(request(
+            self, self.root, "POST", self.url, body=json.dumps({})))
+        self.assertEqual(response.code, 400)
+
+    def test_delete_node_when_node_does_not_exist(self):
+        """
+        Delete node returns 404 when the given node_id
+        does not exist.
+        """
+        response = self.successResultOf(request(
+            self, self.root, "DELETE", self.url + "/1234"))
+        self.assertEqual(response.code, 404)
+
+    def test_delete_node(self):
+        """
+        Delete node returns 204 deletes the given node_id.
+        """
+        # create node
```

Comment:
"create node" comments are unnecessary because the method name is obvious. "delete node" and "get node" might be ok because HTTP noise is cluttering up those operations. Still, I feel the marginal value of these comments is low.